Autopilot/Src/xbee.cpp

```diff
@@ -0,0 +1,43 @@
+#include "xbee.hpp"
+#include "stm32f7xx_hal_usart.h"
+
+extern USART_HandleTypeDef husart2;
+
+XBEE* XBEE::xbeeInstance = nullptr;
+
+XBEE* XBEE::getInstance() {
+    if (!xbeeInstance) {
+        xbeeInstance = new XBEE();
+    }
+
+    return xbeeInstance;
+}
+
+XBEE::XBEE() {}
+
+void XBEE::Receive_GS_Data() {}
+
+void XBEE::Send_GS_Data() {}
+
+xbee_in_t GetResult() {
+	uint8_t holder;
+	xbee_in_t xbeeIn;
+	HAL_USART_Receive(&husart2, &holder, 1, 100);
+	HAL_USART_Receive(&husart2, &xbeeIn.len, 1, 100);
+	uint8_t payload[xbeeIn.len];
```

Comment:
If an array has a variable length, you must allocate it on the heap. Additioanlly, you must make sure you clear this memory so we don't have a memory leak.